libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/trip/model/alert/RouteAlertProvider.kt

```diff
@@ -0,0 +1,54 @@
+package com.mapbox.navigation.base.internal.trip.model.alert
+
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.trip.model.alert.BorderCrossingAdminInfo
+import com.mapbox.navigation.base.trip.model.alert.BorderCrossingAlert
+import com.mapbox.navigation.base.trip.model.alert.RouteAlertGeometry
+import com.mapbox.navigation.base.trip.model.alert.TunnelEntranceAlert
+
+/**
+ * Internal usage prone to breakage.
+ */
+object RouteAlertProvider {
+
+    fun createRouteAlertGeometry(
```

Comment:
what's the benefit of using a provider here? function is exactly a constructor

Reply:
the answer seems to be, to have a way to construct `internal` classes. But can we resolve this with a better package organization?

Reply:
> the answer seems to be, to have a way to construct `internal` classes. But can we resolve this with a better package organization?

"Problem" with this is that it's in `base`, although maybe we should move those models and factories to `core` 🤔